### PR TITLE
Update architecture guidelines with MVP

### PR DESCRIPTION
**Pull request by ivacf**

### CODE REVIEW

File: architecture_guidelines/android_architecture.md [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
+* __Data Manager (Model)__: It's a key part of the architecture. It keeps a reference to every helper class and uses them to satisfy the requests coming from the presenters. Its methods make extensive use of Rx operators to combine, transform or filter the output coming from the helpers in order to generate the desired output ready for the Presenters. It returns observables that emit data models.
+
+* __Presenters__: Subscribe to observables provided by the `DataManager` and process the data in order to call the right method in the View.
+
+* __Activities, Fragments, ViewGroups (View)__: Standard Android components that implement a set of method that the Presenters can call. They also handle user interactions such as clicks and act accordingly by calling the appropriate method in the Presenter. These components also implement framework-related tasks such us managing the Android lifecycle, inflating views, etc.
```
Comment (hitherejoe) [inline architecture_guidelines/android_architecture.md]: methods

Comment (hitherejoe) [inline architecture_guidelines/android_architecture.md]: (first line - a set of method that)

File: architecture_guidelines/android_architecture.md [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
 
-* __Event Bus__: It allows UI layer components to be notified of certain events that happen in the data layer. The data manager posts events which can then be subscribed to by Activities and Fragments.
\ No newline at end of file
+* __Event Bus__: It allows the View components to be notified of certain type of events that happen in the Model. Generally the  `DataManager` posts events which can then be subscribed to by Activities and Fragments. The event bus is __only used for very specific actions__ that are not related to only one screen and have a broadcasting nature, e.g. the user has signed out.
```
Comment (hitherejoe) [inline architecture_guidelines/android_architecture.md]: types of - (certain type of events that happen)

**Comment by hitherejoe** [comment]
Int he diagram, did you purposefully shorter PreferencesHelper to PreferHelper?

**Comment by ivacf** [comment]
Yes I did, the name was too long for the box

**Comment by hitherejoe** [comment]
Ah ok that's a shame, it looked OK in the first image - no worries!

**Pull request merged**